python/langsmith/evaluation/llm_evaluator.py

```diff
@@ -0,0 +1,221 @@
+"""Contains the LLMEvaluator class for building LLM-as-a-judge evaluators."""
+
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union, cast
+
+from pydantic import BaseModel
+
+from langsmith.evaluation import EvaluationResult, EvaluationResults, RunEvaluator
+from langsmith.schemas import Example, Run
+
+
+class CategoricalScoreConfig(BaseModel):
+    """Configuration for a categorical score."""
+
+    key: str
+    choices: List[str]
+    description: str
+    include_explanation: bool = False
+
+
+class ContinuousScoreConfig(BaseModel):
+    """Configuration for a continuous score."""
+
+    key: str
+    min: float = 0
+    max: float = 1
+    description: str
+    include_explanation: bool = False
+
+
+def _create_score_json_schema(
+    score_config: Union[CategoricalScoreConfig, ContinuousScoreConfig],
+) -> dict:
+    properties: Dict[str, Any] = {}
+    if isinstance(score_config, CategoricalScoreConfig):
+        properties["score"] = {
+            "type": "string",
+            "enum": score_config.choices,
+            "description": f"The score for the evaluation, one of "
+            f"{', '.join(score_config.choices)}.",
+        }
+    elif isinstance(score_config, ContinuousScoreConfig):
+        properties["score"] = {
+            "type": "number",
+            "minimum": score_config.min,
+            "maximum": score_config.max,
+            "description": f"The score for the evaluation, between "
+            f"{score_config.min} and {score_config.max}, inclusive.",
+        }
+    else:
+        raise ValueError("Invalid score type. Must be 'categorical' or 'continuous'")
+
+    if score_config.include_explanation:
+        properties["explanation"] = {
+            "type": "string",
+            "description": "The explanation for the score.",
+        }
+
+    return {
+        "title": score_config.key,
+        "description": score_config.description,
+        "type": "object",
+        "properties": properties,
+        "required": (
+            ["score", "explanation"] if score_config.include_explanation else ["score"]
+        ),
+    }
+
+
+class LLMEvaluator(RunEvaluator):
+    """A class for building LLM-as-a-judge evaluators."""
+
+    def __init__(
+        self,
+        *,
+        prompt_template: Union[str, List[Tuple[str, str]]],
+        score_config: Union[CategoricalScoreConfig, ContinuousScoreConfig],
+        map_variables: Optional[Callable[[Run, Optional[Example]], dict]] = None,
+        model: str = "gpt-3.5-turbo",
```

Comment:
Probably shouldn't default to turbo right? Maybe don't even have a default

Reply:
We should probably let them provide a chat model directly also, since langsmith users frequently have their own custom proxy, etc. that wouldn't be supported just with args to init_chat_model

Reply:
I think it helps to have a default, esp if most people are using oai anyways. Will add something to init from chat model.